1_fetch/src/fetch_precip_values.R

```diff
@@ -0,0 +1,77 @@
+
+fetch_precip_values <- function(ind_file, precip_spatial_ind, dates) {
+
+  nc <- ncdf4::nc_open("https://cida.usgs.gov/thredds/dodsC/stageiv_combined")
+
+  precip_spatial <- readRDS(sc_retrieve(precip_spatial_ind))
+
+  t_vals <- get_time_nc(nc$dim$time)
+
+  start_date <- as.Date(dates$start)
+  end_date <- as.Date(dates$end)
+
+  x_inds <- seq(min(precip_spatial$x), max(precip_spatial$x), 1)
+  y_inds <- seq(min(precip_spatial$y), max(precip_spatial$y), 1)
+  t_inds <- which(t_vals > start_date & t_vals < end_date)
+
+  # ensures we make our request in the right axis order.
+  dimid_order <- match(nc$var$Total_precipitation_surface_1_Hour_Accumulation$dimids,
+                       c(nc$dim$x$id, nc$dim$y$id, nc$dim$time$id))
+
+  precip <- array(dim = c(length(x_inds), length(y_inds), length((t_inds))))
+  t_counter <- 1
+
+  for(step in seq(min(t_inds), max(t_inds), 1)) {
+    precip[,,t_counter] <-
+      ncdf4::ncvar_get(nc, nc$var$Total_precipitation_surface_1_Hour_Accumulation,
+                       start <- c(min(x_inds),
+                                  min(y_inds),
+                                  step)[dimid_order],
+                       count <- c(length(x_inds),
+                                  length(y_inds),
+                                  1)[dimid_order])
+    t_counter <- t_counter + 1
+    message(paste(t_counter, length(t_inds)))
```

Comment:
Not worth changing, but the last `message` is `192 191`. Next time maybe do `1 out of 191` and put the `message()` above the counter increment. 

and if you want to give a progress progression, 
```r
if (t_counter %% 10 == 0){
   message(paste(t_counter, 'of', length(t_inds)))
}
```
Will print fewer lines (every ten)